### PR TITLE
Learn Forms design basics

### LINKED ISSUE #5842: Learn Forms: UX / UI
Draft: 
PR:

### PR DESCRIPTION
**Pull request by rachelandrew**
Fixes #5842

### CODE REVIEW

File: src/site/content/en/learn/forms/design-basics/design-basics.assess.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+tabLabel: question
+questions:
+  - type: multiple-choice
+    cardinality: "1"
+    correctAnswers: "1"
+    stem: How do you describe a form control?
+    options:
+      - content: "Using the `<description>` element."
+        rationale: "Try again!"
+      - content: "Using the `<label> element."
```
Comment (justmarkup) [inline src/site/content/en/learn/forms/design-basics/design-basics.assess.yml:11]:
```suggestion
      - content: "Using the `<label>` element."
```

File: src/site/content/en/learn/forms/design-basics/index.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+### How does your user know what data to enter?
+
+Make sure users understand how to enter valid data.
+Do they need to enter at least eight characters? Tell them.
+
+```html
+<label for="password">Password (required)</label>
+<input required minlength="8" type="password" id="password"
+  name="password" aria-describedby="password-minlength">
```
Comment (justmarkup) [inline src/site/content/en/learn/forms/design-basics/index.md:185]:
```suggestion
<input required minlength="8" type="password" id="password" name="password" aria-describedby="password-minlength">
```

File: src/site/content/en/learn/forms/design-basics/index.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+Learn more about the
+[`pointer` CSS media feature](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/pointer).
+
+### Show errors where they happen
+
+To make it straightforward for users to find out which form control needs their attention,
+display error messages next to the form control they refer to.
+When displaying errors on form submission, make sure to navigate to the first invalid form control.
+
+### How does your user know what data to enter?
```
Comment (justmarkup) [inline src/site/content/en/learn/forms/design-basics/index.md:177]: Sounds a bit like you only have one user :-)

```suggestion
### Make it clear to users what data to enter
```

File: src/site/content/en/learn/forms/design-basics/index.md [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+```html
+<label for="password">Password (required)</label>
+<input required minlength="8" type="password" id="password"
+  name="password" aria-describedby="password-minlength">
+<span id="password-minlength">Enter at least eight characters</span>
+```
+
+{% Aside %}
+Make it clear to users which fields are required.
```
Comment (justmarkup) [inline src/site/content/en/learn/forms/design-basics/index.md:190]: Not sure, why I added this inside an aside, I think we don't need the aside here.

```suggestion
### Make it clear to users which fields are required.

File: src/site/content/en/learn/forms/design-basics/index.md [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+<input name="name" id="name">
+```
+
+Learn more about
+[indicating required fields](https://www.deque.com/blog/anatomy-of-accessible-forms-required-form-fields/).
+
+If most fields are required,
+it's also
+[an option to indicate the optional fields](https://www.lukew.com/ff/entry.asp?725).
+{% endAside %}
```
Comment (justmarkup) [inline src/site/content/en/learn/forms/design-basics/index.md:203]:
```suggestion
```

File: src/site/content/en/learn/forms/design-basics/index.md [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+as the `font-size` for your form controls.
+This [prevents a page zoom](https://css-tricks.com/16px-or-larger-text-prevents-ios-form-zoom/)
+on iOS Safari when a form control is focused.
+{% endAside %}
+
+You can define different sizes for different pointing devices,
+for example, a mouse, using the  `pointer` CSS media feature.
+
+```css
+// has a pointing device, for example, a mouse
```
Comment (justmarkup) [inline src/site/content/en/learn/forms/design-basics/index.md:151]:
```suggestion
// pointer device, for example, a mouse
```

File: src/site/content/en/learn/forms/design-basics/index.md [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+@media (pointer: fine) {
+  input[type="checkbox"] {
+    width: 15px;
+    height: 15px;
+  }
+}
+
+// pointing device of limited accuracy, for example, a touch-based device
```
Comment (justmarkup) [inline src/site/content/en/learn/forms/design-basics/index.md:159]:
```suggestion
// pointer device of limited accuracy, for example, a touch-based device
```

**Comment by justmarkup** [comment]
@googlebot I fixed it.

**Comment by samdutton** [comment]
Hi @justmarkup — looks like there's a CLA for you, but you might have to post a comment with `@googlebot I fixed it`.

**Comment by samdutton** [comment]
@justmarkup All good now after rescan.

**Review by samdutton — APPROVED** [review]

**Review by justmarkup — APPROVED** [review]

**Review by robdodson — APPROVED** [review]

**Pull request merged**